gapic/samplegen/samplegen.py

```diff
@@ -141,25 +246,45 @@ def inner_validate_(self, response):
 
             validater(self, body)
 
-    def validate_comment_(self, body: List[str]):
-        fmt_str = body[0]
-        num_vars = fmt_str.count("%s")
-        if num_vars != len(body) - 1:
-            raise MismatchedPrintStatement(
-                "Expected {} var names in comment but received {}"
-                .format(num_vars, len(body) - 1))
-
-        for var_name in body[1:]:
-            var_base = var_name.split(".")[0]
-            if var_base not in self.var_defs_:
-                raise UndefinedVariableReference("Reference to undefined variable: {}"
-                                                 .format(var_base))
+        return True
+
+    def _handle_lvalue(self, lval):
+        """Internal helper method that does safety checks on an lvalue
+           and adds it to lexical scope.
 
-    def validate_print_(self, body: List[str]):
+        Raises:
+            ReservedVariableName: If an attempted lvalue is a reserved keyword.
+        """
+        if lval in RESERVED_WORDS:
+            raise ReservedVariableName(
+                "Tried to define a variable with reserved name: {}".format(lval))
+
+        # Even though it's valid python to reassign variables to any rvalue,
+        # the samplegen spec prohibits this.
+        if lval in self.var_defs_:
+            raise RedefinedVariable(
+                "Tried to redefine variable: {}".format(lval))
+
+        self.var_defs_.add(lval)
+
+    def _validate_format(self, body: List[str]):
+        """Validates a format string and corresponding arguments.
+
+         The number of format tokens in the string must equal the
+         number of arguments, and each argument must be a defined variable.
+
+         TODO: the attributes of the variable must correspond to attributes
+               of the variable's type.        
+
+         Raises:
+             MismatchedFormatSpecifier: If the number of format string segments ("%s") in
+                                        a "print" or "comment" block does not equal the
+                                        size number of strings in the block minus 1.
+        """
         fmt_str = body[0]
         num_prints = fmt_str.count("%s")
         if num_prints != len(body) - 1:
-            raise MismatchedPrintStatement(
+            raise MismatchedFormatSpecifier(
```

Comment:
"in format string"

Reply:
To be explicit, I meant in the text of the exception, since this is raised not just for `print` statements